### PR TITLE
Added PCA decomposition plot

### PR DESCRIPTION
**Pull request by georgia-adam**
Completes issue #96

### CODE REVIEW

**Comment by idomic** [comment]
@georgia-adam Looks like the [CI is failing](https://github.com/ploomber/sklearn-evaluation/actions/runs/3809510183/jobs/6485738166), looks like linting issues, we're using black for that, make sure it's passing before we can review.

**Comment by idomic** [comment]
Also the plot is missing a few data points on one of the classes and is missing an index (compared to the original one in the issue).

File: src/sklearn_evaluation/plot/pca.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import StandardScaler
+from ..telemetry import SKLearnEvaluationLogger
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:3]: Change this to : `from sklearn_evaluation.telemetry import SKLearnEvaluationLogger`

File: src/sklearn_evaluation/plot/pca.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+               , s = 50)
+        
+    _set_ax_settings(ax,targets)
+    
+    return ax
+
+
+def _set_ax_settings(ax,targets=None):
+    ax.set_title('Principal Component Plot', fontsize = 10)
+#     ax.set_xlim([-2, 2])
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:144]: Please remove commented code

File: src/sklearn_evaluation/plot/pca.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            n_components == min(n_samples, n_features) - 1
+    
+    standard_scalar : Flag 
+        set True if features needs to be scaled
+    
+    ax : matplotlib Axes
+        Axes object to draw the plot onto, otherwise uses current Axes
+
+    Notes
+    -----
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:52]: Add z `..versionadded` tag (check contribution guideline)

File: src/sklearn_evaluation/plot/pca.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    ax: matplotlib Axes
+        Axes containing the plot
+
+    Examples
+    --------
+    .. plot:: ../examples/pca.py
+
+    """
+    if np.isnan(X).sum():
+        raise ValueError('X array should not consist nan '
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:67]: `consist` -> `contain`

File: tests/test_plot.py [hunk 5]
```diff
@@ -118,6 +118,15 @@ def test_roc_multi_add_to_multi(
...
 
 
+@image_comparison(baseline_images=["pca"])
+def test_pca():
+    weights = np.array([73,95,72,69,76,82,95,89,89,83,94,89,103]) 
+    heights = np.array([56,73,60,54,56,62,77,73,72,69,74,67,84,]) 
+    X = np.array((heights,weights)).transpose()
+
+    plot.pca(X,standard_scalar=True)
+
```
Comment (neelasha23) [inline tests/test_plot.py:128]: Please add some test case, e.g., 
1. Different input type like pandas
2. Create a dummy dataset and then run pca on it
3. check that passing when `ax` is passed the plot is not creating a new axes

File: src/sklearn_evaluation/plot/pca.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    if standard_scalar:
+        X = StandardScaler().fit_transform(X)
+    
+    # PCA Projection to 2D
+    _pca(X, Y, n_components, ax)
+
+    return ax
+
+
+def _pca(X, Y=None, n_components=2, ax=None):
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:83]: There's a method named `pca` already, consider renaming this to something more specific

File: src/sklearn_evaluation/plot/pca.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        
+    if ax is None:
+        ax = plt.gca()
+    
+    # Standardizing the features
+    if standard_scalar:
+        X = StandardScaler().fit_transform(X)
+    
+    # PCA Projection to 2D
+    _pca(X, Y, n_components, ax)
```
Comment (neelasha23) [inline src/sklearn_evaluation/plot/pca.py:78]: `return  _pca(X, Y, n_components, ax)` should also work

**Comment by neelasha23** [comment]
Getting this error `ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().` when dataset is read as a Pandas dataframe.

**Pull request closed without merging**